Lib/ntpath.py

```diff
@@ -311,6 +311,39 @@ def ismount(path):
         return False
 
 
+_reserved_chars = frozenset(
+    {chr(i) for i in range(32)} |
+    {'"', '*', ':', '<', '>', '?', '|'}
+)
+
+_reserved_names = frozenset(
+    {'CON', 'PRN', 'AUX', 'NUL', 'CONIN$', 'CONOUT$'} |
+    {f'COM{c}' for c in '123456789\xb9\xb2\xb3'} |
+    {f'LPT{c}' for c in '123456789\xb9\xb2\xb3'}
+)
+
+def isreserved(path):
+    """Return true if the pathname is reserved by the system."""
+    # Refer to "Naming Files, Paths, and Namespaces":
+    # https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    path = splitdrive(path)[1]
+    path = os.fsdecode(path).replace(altsep, sep)
+    for name in path.split(sep):
+        # Trailing dots and spaces are reserved.
+        if name.endswith(('.', ' ')) and name not in ('.', '..'):
+            return True
```

Comment:
What do you think about returning `name` instead of `True`? It's still boolean true, and it's helpful to know which component is the problem.

Reply:
I'm ambivalent. I'd prefer that we add an `isreservedname()` function that users could call if they need that level of granularity. It would give them more options too, e.g. they could collect _all_ the invalid path segments rather than the first one.

Reply:
Okay, so do you just prefer it over my weird idea, or are you really in favor moving the body of the loop into `isreservedname()`? It's fine by me.

Reply:
I suppose the path function would be better named `isreservedpath()` in that case.

Reply:
For the sake of discussion I've added `isreservedname()`. No tests/docs yet as I'm not sure what you and @zooba think of the idea - can revert if we don't like it.

I quite like it, but I'm an amateur :)

Reply:
I'd rather stick to returning a boolean. I can see the value of returning more info, but not through as `is*` API (one easy example is that users might print the result, which would be very confusing to not see "True", but there are other more subtle ways for it to get weird).

However, I don't see why trailing dots and spaces are reserved? They'll be normalised out most of the time, which means the path may well be real (I assume they're ignored for the purpose of reserved names - `C:\NUL .` is still not going to be a user-creatable file). So probably `isreservedname` should strip them.

Posting a comment below about `isreservedname`

Reply:
> However, I don't see why trailing dots and spaces are reserved?

Such names can't be created or accessed by normal means. It isn't fundamentally different from how `open('con')` actually opens "\\\\.\\con" instead of a file named "con" in the current working directory [^1]. 

Some reserved names can be created by using an extended path, but that makes them proper NT paths[^2], not proper Windows paths. This helps to deal with someone else's misuse of extended paths, as a method of last resort, but it's not a license to create such nuisance filenames.

[^1]: This why all DOS device names still have to be reserved in Windows 11. If one can't access a bare name relative to the current working directory without an explicit workaround such as ".\\con", then it's too magical for normal use. Eventually we can stop reserving device names that have an extension such as "con.txt", once older Windows systems are so rarely used that they're not worth considering.
[^2]: Rather, they become proper NT paths after the system replaces "\\\\?\\" with NT's local device directory "\\??\\".

Reply:
I would stick with returning a boolean. If someone shows a good need for returning the reserved word that was found we can change it later.